flash/core/integrations/labelstudio/data_source.py

```diff
@@ -0,0 +1,266 @@
+    Export data should point to image files"""
+
+    def load_sample(self, sample: Mapping[str, Any] = None, dataset: Optional[Any] = None) -> Any:
+        """Load 1 sample from dataset."""
+        p = sample["file_upload"]
+        # loading image
+        image = default_loader(p)
+        result = {DefaultDataKeys.INPUT: image, DefaultDataKeys.TARGET: self._get_labels_from_sample(sample["label"])}
+        return result
+
+
+class LabelStudioTextClassificationDataSource(LabelStudioDataSource):
+    """The ``LabelStudioTextDataSource`` expects the input to
+    :meth:`~flash.core.data.data_source.DataSource.load_data` to be a json export from label studio.
+    Export data should point to text data
+    """
+
+    def __init__(self, backbone=None, max_length=128):
+        super().__init__()
+        if backbone:
+            if _TEXT_AVAILABLE:
+                from transformers import AutoTokenizer
+            self.backbone = backbone
+            self.tokenizer = AutoTokenizer.from_pretrained(backbone, use_fast=True)
+            self.max_length = max_length
+
+    def load_sample(self, sample: Mapping[str, Any] = None, dataset: Optional[Any] = None) -> Any:
+        """Load 1 sample from dataset."""
+        if self.backbone:
+            data = ""
+            for key in sample.get("data"):
+                data += sample.get("data").get(key)
+            tokenized_data = self.tokenizer(data, max_length=self.max_length, truncation=True, padding="max_length")
+            for key in tokenized_data:
+                tokenized_data[key] = torch.tensor(tokenized_data[key])
+            tokenized_data["labels"] = self._get_labels_from_sample(sample["label"])
+            # separate text data type block
+            result = tokenized_data
+        return result
+
+
+class LabelStudioVideoClassificationDataSource(LabelStudioDataSource):
+    """The ``LabelStudioVideoDataSource`` expects the input to
+    :meth:`~flash.core.data.data_source.DataSource.load_data` to be a json export from label studio.
+    Export data should point to video files"""
+
+    def __init__(self, video_sampler=None, clip_sampler=None, decode_audio=False, decoder: str = "pyav"):
+        if not _PYTORCHVIDEO_AVAILABLE:
+            raise ModuleNotFoundError("Please, run `pip install pytorchvideo`.")
+        super().__init__()
+        self.video_sampler = video_sampler or torch.utils.data.RandomSampler
+        self.clip_sampler = clip_sampler
+        self.decode_audio = decode_audio
+        self.decoder = decoder
+
+    def load_sample(self, sample: Mapping[str, Any] = None, dataset: Optional[Any] = None) -> Any:
+        """Load 1 sample from dataset."""
+        return sample
+
+    def load_data(self, data: Optional[Any] = None, dataset: Optional[Any] = None) -> Sequence[Mapping[str, Any]]:
+        """load_data produces a sequence or iterable of samples."""
+        res = super().load_data(data, dataset)
+        return self.convert_to_encodedvideo(res)
```

Comment:
How does this work with the stuff attached to self? There you could end up with a dataset that hasn't been wrapped. Possible solution would be to wrap the test set and val set here if they were set in the super?

Reply:
This work like that:
1. Load data with default load_data in LabelStudioDataSource
2. Wrap result with convert_to_encodedvideo method 

Could you please suggest the scenario for checking the situation with the unwrapped dataset?